# Laboratorio 11 — Distribución normal y puntuación Z

En el laboratorio anterior observamos que la variable `peso_producto`
presenta una distribución aproximadamente normal.

En este laboratorio utilizaremos esta variable para aplicar los conceptos
estudiados en teoría:

- regla 68–95–99.7;
- distancia respecto a la media;
- puntuación Z;
- interpretación de valores por encima y por debajo de la media.

### Objetivos

- Comprobar empíricamente la regla 68–95–99.7.
- Determinar qué proporción de los datos se encuentra a una, dos y tres
  desviaciones estándar de la media.
- Calcular la puntuación Z de observaciones individuales.
- Interpretar el signo y la magnitud de una puntuación Z.

## 1. Carga de la base de datos

Trabajaremos nuevamente con `datos_distribuciones.csv`.

Para este laboratorio nos concentraremos en la variable `peso_producto`,
que fue simulada a partir de una distribución Normal.

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

datos = pd.read_csv("/content/datos_distribuciones.csv")

datos.head()

,id,compras_exitosas,errores_hora,tiempo_respuesta,peso_producto
0,1,7,5,0.080113,546.827611
1,2,3,4,7.569356,524.451797
2,3,5,2,0.555077,500.940658
3,4,6,5,2.177190,493.162475
4,5,8,8,3.653941,504.966896


## 2. Media y desviación estándar

Antes de aplicar la regla 68–95–99.7 necesitamos conocer el centro
y la dispersión de la variable `peso_producto`.

Calcularemos nuevamente su media y su desviación estándar.

In [2]:
media = datos["peso_producto"].mean()
desviacion = datos["peso_producto"].std()
print(f"Media: {media:.2f} g")
print(f"Desviación estándar: {desviacion:.2f} g")

Media: 498.67 g
Desviación estándar: 24.99 g


In [3]:
limite_inferior_1 = media - desviacion
limite_superior_1 = media + desviacion
print(f"Límite inferior: {limite_inferior_1:.2f} g")
print(f"Límite superior: {limite_superior_1:.2f} g")

Límite inferior: 473.68 g
Límite superior: 523.66 g


## 4. Comprobación de la regla del 68 %

Según la regla 68–95–99.7, en una distribución aproximadamente normal
esperamos que cerca del **68 % de los valores** se encuentre entre:

$
\mu-\sigma \leq X \leq \mu+\sigma
$

Para nuestra variable `peso_producto`, este intervalo es aproximadamente:

$
473.68 \leq X \leq 523.66
$

Vamos a comprobar qué porcentaje de las observaciones de nuestra base
se encuentra realmente dentro de este intervalo.

In [4]:
dentro_1_sigma = datos[
    (datos["peso_producto"] >= limite_inferior_1) &
    (datos["peso_producto"] <= limite_superior_1)
]
cantidad_1_sigma = len(dentro_1_sigma)
porcentaje_1_sigma = (
    cantidad_1_sigma / len(datos)
) * 100
print(f"Observaciones dentro de ±1σ: {cantidad_1_sigma}")
print(f"Porcentaje observado: {porcentaje_1_sigma:.2f}%")
print("Porcentaje teórico aproximado: 68%")

Observaciones dentro de ±1σ: 702
Porcentaje observado: 70.20%
Porcentaje teórico aproximado: 68%


### Interpretación

A partir del resultado obtenido, responda:

1. ¿Qué porcentaje de los datos se encuentra entre una desviación estándar
   por debajo y una desviación estándar por encima de la media?

2. ¿El porcentaje obtenido es exactamente 68 %?

3. ¿El resultado es coherente con la regla 68–95–99.7? Justifique.

In [5]:
limite_inferior_2 = media - 2 * desviacion
limite_superior_2 = media + 2 * desviacion

dentro_2_sigma = datos[
    (datos["peso_producto"] >= limite_inferior_2) &
    (datos["peso_producto"] <= limite_superior_2)
]

cantidad_2_sigma = len(dentro_2_sigma)
porcentaje_2_sigma = (cantidad_2_sigma / len(datos)) * 100

print(f"Intervalo: {limite_inferior_2:.2f} g a {limite_superior_2:.2f} g")
print(f"Observaciones dentro de ±2σ: {cantidad_2_sigma}")
print(f"Porcentaje observado: {porcentaje_2_sigma:.2f}%")
print("Porcentaje teórico aproximado: 95%")

Intervalo: 448.69 g a 548.65 g
Observaciones dentro de ±2σ: 944
Porcentaje observado: 94.40%
Porcentaje teórico aproximado: 95%


### Interpretación

A partir de los resultados obtenidos, responda:
1. ¿Cuál es el intervalo comprendido entre \(\mu-2\sigma\) y \(\mu+2\sigma\)?
2. ¿Qué porcentaje de las observaciones se encuentra dentro de este intervalo?
3. ¿El porcentaje observado es cercano al 95 % esperado?
4. Compare este resultado con el obtenido para \(\pm1\sigma\). ¿Por qué ahora se incluye una mayor cantidad de observaciones?

## 6. Comprobación de la regla del 99.7 %

Finalmente, ampliaremos el intervalo a **tres desviaciones estándar**
a cada lado de la media:

\[
\mu-3\sigma \leq X \leq \mu+3\sigma
\]

Según la regla 68–95–99.7, aproximadamente el **99.7 % de los valores**
de una distribución Normal se encuentra dentro de este intervalo.

Comprobaremos qué porcentaje obtenemos en nuestra base de datos.

In [6]:
limite_inferior_3 = media - 3 * desviacion
limite_superior_3 = media + 3 * desviacion
dentro_3_sigma = datos[
    (datos["peso_producto"] >= limite_inferior_3) &
    (datos["peso_producto"] <= limite_superior_3)
]
cantidad_3_sigma = len(dentro_3_sigma)
porcentaje_3_sigma = (cantidad_3_sigma / len(datos)) * 100
print(f"Intervalo: {limite_inferior_3:.2f} g a {limite_superior_3:.2f} g")
print(f"Observaciones dentro de ±3σ: {cantidad_3_sigma}")
print(f"Porcentaje observado: {porcentaje_3_sigma:.2f}%")
print("Porcentaje teórico aproximado: 99.7%")

Intervalo: 423.70 g a 573.64 g
Observaciones dentro de ±3σ: 993
Porcentaje observado: 99.30%
Porcentaje teórico aproximado: 99.7%


### Interpretación

A partir de los resultados obtenidos, responda:

1. ¿Cuál es el intervalo comprendido entre \(\mu-3\sigma\) y \(\mu+3\sigma\)?
2. ¿Qué porcentaje de las observaciones se encuentra dentro de este intervalo?
3. ¿El resultado es cercano al 99.7 % esperado?
4. ¿Qué ocurrió con el porcentaje de observaciones al pasar de \(\pm1\sigma\) a \(\pm2\sigma\) y finalmente a \(\pm3\sigma\)?

## 7. Comparación con la regla 68–95–99.7

Ya calculamos la proporción de observaciones que se encuentra a una,
dos y tres desviaciones estándar de la media.

Ahora compararemos los porcentajes obtenidos en nuestra base de datos
con los valores teóricos de la regla 68–95–99.7.

In [7]:
comparacion = pd.DataFrame({
    "Intervalo": ["±1σ", "±2σ", "±3σ"],
    "Porcentaje observado": [
        porcentaje_1_sigma,
        porcentaje_2_sigma,
        porcentaje_3_sigma
    ],
    "Porcentaje teórico": [
        68.0,
        95.0,
        99.7
    ]
})

comparacion

,Intervalo,Porcentaje observado,Porcentaje teórico
0,±1σ,70.2,68.0
1,±2σ,94.4,95.0
2,±3σ,99.3,99.7


### Análisis de los resultados

Observe la tabla comparativa y responda:

1. ¿Los porcentajes observados son exactamente iguales a los porcentajes teóricos?

2. ¿Los resultados obtenidos son cercanos a los valores esperados por la regla 68–95–99.7?

3. ¿Qué ocurre con la proporción de observaciones incluidas cuando aumentamos de una a tres desviaciones estándar?

4. A partir de los resultados obtenidos, ¿considera que el comportamiento de `peso_producto` es coherente con una distribución aproximadamente normal? Justifique.

## 8. Cálculo de la puntuación Z

Hasta ahora analizamos intervalos alrededor de la media.

Ahora queremos determinar la posición de **cada observación individual**
respecto a la media utilizando la puntuación Z.

Recordemos:

$
Z=\frac{x-\mu}{\sigma}
$

El valor de Z indica cuántas desviaciones estándar se encuentra una
observación por encima o por debajo de la media.

In [8]:
datos["z_score"] = (
    (datos["peso_producto"] - media) / desviacion
)
datos[["peso_producto", "z_score"]].head()

,peso_producto,z_score
0,546.827611,1.927256
1,524.451797,1.031857
2,500.940658,0.091026
3,493.162475,-0.220228
4,504.966896,0.252142


In [9]:
datos[["peso_producto", "z_score"]].head(10)

,peso_producto,z_score
0,546.827611,1.927256
1,524.451797,1.031857
2,500.940658,0.091026
3,493.162475,-0.220228
4,504.966896,0.252142
5,512.938002,0.571116
6,449.174982,-1.980449
7,552.808743,2.166599
8,480.252892,-0.736823
9,499.952359,0.051478


### Interpretación

Observe los valores de `peso_producto` y su correspondiente `z_score`.

Responda:

1. ¿Qué ocurre con el signo de Z cuando el peso está por encima de la media?

2. ¿Qué ocurre con el signo de Z cuando el peso está por debajo de la media?

3. ¿Qué significa encontrar una observación con un valor de Z cercano a 0?

4. ¿Qué observación de las primeras diez parece encontrarse más alejada de la media? Justifique utilizando su Z-score.

## 9. Observaciones más alejadas de la media

La magnitud del Z-score permite determinar qué tan lejos se encuentra
una observación de la media.

- Un Z positivo indica un valor por encima de la media.
- Un Z negativo indica un valor por debajo de la media.
- Cuanto mayor sea $|Z|$, mayor será la distancia respecto a la media.

Buscaremos los productos con el Z-score más bajo y más alto de nuestra base.

In [10]:
indice_min = datos["z_score"].idxmin()
indice_max = datos["z_score"].idxmax()
observacion_min = datos.loc[
    indice_min,
    ["id", "peso_producto", "z_score"]
]
observacion_max = datos.loc[
    indice_max,
    ["id", "peso_producto", "z_score"]
]
print("Observación con menor Z-score:")
print(observacion_min)
print("\nObservación con mayor Z-score:")
print(observacion_max)

Observación con menor Z-score:
id               648.000000
peso_producto    410.448435
z_score           -3.530145
Name: 647, dtype: float64

Observación con mayor Z-score:
id               291.000000
peso_producto    575.367793
z_score            3.069331
Name: 290, dtype: float64
